src/ng/filter/filter.js

```diff
@@ -196,7 +196,7 @@ function deepCompare(actual, expected, comparator, matchAgainstAnyProp) {
       } else if (expectedType === 'object') {
         for (key in expected) {
           var expectedVal = expected[key];
-          if (isFunction(expectedVal)) {
+          if (isFunction(expectedVal) || isUndefined(expectedVal)) {
```

Comment:
@pkozlowski-opensource here it is: https://github.com/angular/angular.js/commit/63b9956faf4c3679c88a9401b8ccbb111c0294ee